### PR TITLE
nixos/doc: convert boot problems section to CommonMark

### PR DESCRIPTION
**Pull request by ryantm**
###### Motivation for this change
 [RFC 72](https://github.com/NixOS/rfcs/blob/master/rfcs/0072-commonmark-docs.md)

### CODE REVIEW

File: nixos/doc/manual/administration/boot-problems.section.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+`single`
+
+: Boot into rescue mode (a.k.a. single user mode). This will cause systemd to start nothing but the unit `rescue.target`, which runs `sulogin` to prompt for the root password and start a root login shell. Exiting the shell causes the system to continue with the normal boot process.
+
+`systemd.log_level=debug systemd.log_target=console`
+
+: Make systemd very verbose and send log messages to the console instead of the journal.
+
+: For more parameters recognised by systemd, see systemd(1).
```
Comment (jtojnar) [inline nixos/doc/manual/administration/boot-problems.section.md:33]: This does not belong to the definition list.

File: nixos/doc/manual/administration/boot-problems.section.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+: Start an interactive shell in stage 1 before anything useful has been done. That is, no modules have been loaded and no file systems have been mounted, except for `/proc` and `/sys`.
+
+`boot.debug1devices`
+
+: Like `boot.debug1`, but runs stage1 until kernel modules are loaded and device nodes are created. This may help with e.g. making the keyboard work.
+
+`boot.debug1mounts`
+
+: Like `boot.debug1` or `boot.debug1devices`, but runs stage1 until all filesystems that are mounted during initrd are mounted (see [neededForBoot](options.html#opt-fileSystems._name_.neededForBoot)). As a motivating example, this could be useful if you've forgotten to set [neededForBoot](options.html#opt-fileSystems._name_.neededForBoot) on a file system.
```
Comment (jtojnar) [inline nixos/doc/manual/administration/boot-problems.section.md:19]: You should just use `#opt-fileSystems._name_.neededForBoot`, see https://github.com/NixOS/nixpkgs/issues/102246#issuecomment-854457067

File: nixos/doc/manual/administration/boot-problems.section.md (hunk 1 above)
Comment (ryantm) [reply nixos/doc/manual/administration/boot-problems.section.md:33]: I know what I have doesn't match the original, but I feel like this paragraph only applies to that systemd item, not the rest of the text.

Comment (ryantm) [reply nixos/doc/manual/administration/boot-problems.section.md]: I merged into one definition.

File: nixos/doc/manual/administration/boot-problems.section.md (hunk 2 above)
Comment (ryantm) [reply nixos/doc/manual/administration/boot-problems.section.md]: Great, fixed!

**Comment by ryantm** [comment]
Rendered:
image

**Pull request merged**